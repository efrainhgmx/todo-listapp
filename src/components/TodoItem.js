import React from 'react'

export const TodoItem = ({ todo, toggleTodo }) => {
    const {id, task, completed } = todo;

    const handleTodoClick = () => {
        toggleTodo(id);
    }

    return (
        <li>
            <input type="checkbox" id={ id } checked={ completed }  onChange={ handleTodoClick }/>
            <label for={ id }>{ task }</label>
        </li>
    )
}
