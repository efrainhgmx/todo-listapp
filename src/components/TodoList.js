import React from 'react';

export const TodoList = ({ todos }) => {
    return (
        <ul>
            { todos.map((todo) => (
                <li>Tarea</li>
            )) }
        </ul>
    )
}
