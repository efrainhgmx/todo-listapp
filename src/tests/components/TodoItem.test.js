import React  from 'react';
import { v4 as uuidv4 } from 'uuid';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { TodoItem } from '../../components/TodoItem';

describe('Test on <TodoItem /> component', () => {
    test('Id should be the same as task', () => {
        const list = [{id: uuidv4(), task: "Tarea", completed: false }];
        const key = list.id;
        const wrapper = shallow( <TodoItem key={ key } todo={ list }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( key ).toBe( list.id );
    })
    
})
