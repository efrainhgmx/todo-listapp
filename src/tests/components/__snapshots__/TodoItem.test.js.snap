// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <TodoItem /> component Id should be the same as task 1`] = `
<li>
  <input
    onChange={[Function]}
    type="checkbox"
  />
  <label />
</li>
`;
